### ISSUE TITLE
"TypeError: is_state() takes 3 positional arguments but 4 were given"  When calling trigger with arguments

### ISSUE THREAD

**Issue by Arkanayan** [pytransitions/transitions#314]
The code:
```python

class Clause:
    def __init__(self):#, numString, text, ilvl=None, num_id=None):
        # self.numString = numString
        # self.text = text
        # self.ilvl = ilvl
        # self.num_id = num_id
        self.numString = None
        self.text = None
        self.ilvl = None
        self.num_id = None

    def __repr__(self):
        return '<Clause: numString: {}, text: {}, ilvl: {}, num_id: {}>'\
            .format(self.numString, self.text, self.ilvl, self.num_id)

    def is_first_level(self, paragraph):
        # import ipdb; ipdb.set_trace()
        try:
            if re.match("^\d\.?\d?\.?\d?\.?$", paragraph['numString'].strip()):
                return True
        except Exception as e:
            return False

        return False
    
    def enter_first_level(self, event):
        print('First level entered')    


clause = Clause()
    states = ['first_level']
    transitions = [
        {'trigger': 'process', 'source': 'undecided', 'dest': 'first_level', 'conditions': ['is_first_level'], 'after': 'enter_first_level'}
    ]
    machine = Machine(model=clause, initial='undecided', send_event=True)
    machine.add_states(states)
    machine.add_transitions(transitions)

    for ind, paragraph in enumerate(document_json):
        clause.process(paragraph)
```

I am getting this exception 
```python
  File "parse_json_fsm.py", line 55, in <module>
    clause.process(paragraph)
  File "/home/user/miniconda3/envs/generic_env/lib/python3.5/site-packages/transitions/core.py", line 383, in trigger
    return self.machine._process(func)
  File "/home/user/miniconda3/envs/generic_env/lib/python3.5/site-packages/transitions/core.py", line 1046, in _process
    return trigger()
  File "/home/user/miniconda3/envs/generic_env/lib/python3.5/site-packages/transitions/core.py", line 399, in _trigger
    return self._process(event_data)
  File "/home/user/miniconda3/envs/generic_env/lib/python3.5/site-packages/transitions/core.py", line 409, in _process
    if trans.execute(event_data):
  File "/home/user/miniconda3/envs/generic_env/lib/python3.5/site-packages/transitions/core.py", line 254, in execute
    if not cond.check(event_data):
  File "/home/user/miniconda3/envs/generic_env/lib/python3.5/site-packages/transitions/core.py", line 181, in check
    return predicate(event_data) == self.target
TypeError: is_state() takes 3 positional arguments but 4 were given
```
Is there something wrong with my code? or is this a bug?

**Comment by aleneum** [comment]
Hello @Arkanayan,

as mentioned in the documentation about [checking states](https://github.com/pytransitions/transitions#checking-state), transitions will add convenience methods to your model according to the pattern `is_<<state_name>>`. The method `Clause.is_first_level` matches this pattern and is overridden.
If you change the method's name to `matches_first_level`, your code runs fine.

Even though the overriding is currently expected behaviour, I consider skipping binding convenience functions if a method is already present. However, for your current situation I would recommend renaming your methods to prevent collisions.

**Comment by aleneum** [comment]
Fixed by e60bc1e. The shown 'fail' is just travis lacking Python 3.7 support.
